src/Shared/CertificateGeneration/UnixCertificateManager.cs

```diff
@@ -47,26 +130,707 @@ protected override X509Certificate2 SaveCertificateCore(X509Certificate2 certifi
     internal override CheckCertificateStateResult CheckCertificateState(X509Certificate2 candidate, bool interactive)
     {
         // Return true as we don't perform any check.
+        // This is about checking storage, not trust.
         return new CheckCertificateStateResult(true, null);
     }
 
     internal override void CorrectCertificateState(X509Certificate2 candidate)
     {
         // Do nothing since we don't have anything to check here.
+        // This is about correcting storage, not trust.
     }
 
     protected override bool IsExportable(X509Certificate2 c) => true;
 
-    protected override void TrustCertificateCore(X509Certificate2 certificate) =>
-        throw new InvalidOperationException("Trusting the certificate is not supported on linux");
+    protected override TrustLevel TrustCertificateCore(X509Certificate2 certificate)
+    {
+        var homeDirectory = Environment.GetFolderPath(Environment.SpecialFolder.UserProfile);
+
+        // Rather than create a temporary file we'll have to clean up, we prefer to export the dev cert
+        // to its final location in the OpenSSL directory.  As a result, any failure up until that point
+        // is fatal (i.e. we can't trust the cert in other locations).
+
+        var certDir = GetOpenSslCertificateDirectory(homeDirectory)!; // May not exist
+
+        var nickname = GetCertificateNickname(certificate);
+        var certPath = Path.Combine(certDir, nickname) + ".pem";
+
+        var needToExport = true;
+
+        // We do our own check for file collisions since ExportCertificate silently overwrites.
+        if (File.Exists(certPath))
+        {
+            try
+            {
+                var existingCert = new X509Certificate2(certPath);
+                if (!existingCert.RawDataMemory.Span.SequenceEqual(certificate.RawDataMemory.Span))
```

Comment:
Isn't it enough to compare the thumbprints?

Reply:
I've been reviewing the other implementations and noticed that they do that.  I'm happy to adopt that approach if that's the standard approach.  I only went further because I didn't think SHA1 was considered sufficient for anything security related.